dali/operators/reader/video_reader_decoder_cpu_op.cc

```diff
@@ -32,16 +33,28 @@ void VideoReaderDecoderCpu::RunImpl(SampleWorkspace &ws) {
   video_output.Copy(sample.data_);
   video_output.SetSourceInfo(sample.data_.GetSourceInfo());
 
+  int out_index = 1;
   if (has_labels_) {
-    auto &label_output = ws.Output<CPUBackend>(1);
+    auto &label_output = ws.Output<CPUBackend>(out_index);
     label_output.Resize({}, DALIDataType::DALI_INT32);
     label_output.mutable_data<int>()[0] = sample.label_;
+    out_index++;
+  }
+  if (has_frame_no_) {
+    auto &frame_no_output = ws.Output<CPUBackend>(out_index);
+    frame_no_output.Resize({}, DALIDataType::DALI_INT32);
+    frame_no_output.mutable_data<int>()[0] = sample.first_frame_;
+    out_index++;
   }
 }
 
 namespace detail {
 inline int VideoReaderDecoderOutputFn(const OpSpec &spec) {
-  return spec.HasArgument("labels") ? 2 : 1;
+  int num_outputs = 1;
+  if (spec.HasArgument("labels")) num_outputs++;
+  bool enable_frame_num = spec.GetArgument<bool>("enable_frame_num");
+  if (enable_frame_num) num_outputs++;
+  return num_outputs;
```

Comment:
Perhaps clearer (?) Your call.
```suggestion
  bool has_label_output = spec.HasArgument("labels");
  bool has_frame_num_output = spec.GetArgument<bool>("enable_frame_num");
  return 1 + has_label_output + has_frame_num_output;
```

Reply:
Done